### DISCUSSION TITLE
Подсказка в оглавлении

### DISCUSSION THREAD

**Discussion by furtivite** [doka-guide/platform#355] [Идеи]
## Проблема

У нас есть несколько статей, которые имеют одинаковое название, например

- [.toString()](https://doka.guide/js/number-tostring/) (на самом деле это `number.toString()`)
- [.toString()](https://doka.guide/js/object-tostring/) (на самом деле это `Object.toString()`)

Если смотреть в алфавитную выдачу получается непонятно что и куда

![Снимок экрана 2021-09-14 080418](https://user-images.githubusercontent.com/50330458/133198010-85d7b4f6-722f-4f77-b116-cdf8c40c1e83.png)

но при это очевидно, что в выдаче по темам, всё очень логично расположено:

![Снимок экрана 2021-09-14 080544](https://user-images.githubusercontent.com/50330458/133198130-8540093e-1c89-4e7a-96a1-b0e27f61c9bb.png)

и, тем не менее, возникает вопрос, почему у массы статей про `Element.*` он указан (хотя не везде)?

![Снимок экрана 2021-09-14 080728](https://user-images.githubusercontent.com/50330458/133198323-7e2cd977-be15-4338-9f8b-068a7d8aff85.png)
 
## Предлагается 

1. в статьях, где авторы и лиды направления не хотят писать полные названия в виде `number.toString()` и `Object.toString()` продолжать указывать `.toString()` и `.toString()`, соответственно
1. добавить в статьи опциональное поле `parent` или `hint`, чтобы получить 
    ```njk
    ---
    title: "`.toString()`"
    parent: "`number`"
    ```

    или 

    ```njk
    ---
    title: "`.toString()`"
    hint: "`Object`"
    ```

    Это позволит сохранить организацию по алфавиту на букву "Т"
1. Выводить их в оглавлениях по-алфавиту (кажется, это не требуется в выводе группами, потому что и так понятно)
варианты внешнего вида:
    акцентным цветом
    ![Снимок экрана 2021-09-14 082018](https://user-images.githubusercontent.com/50330458/133199391-e1579694-4d9d-4506-b019-aaee87a27d51.png)
    просто серым
    ![Снимок экрана 2021-09-14 082336](https://user-images.githubusercontent.com/50330458/133199665-2ef0c93f-629c-4301-8baf-b404a2bed072.png)
    Решение принимается командой дизайнеров. Изображения даны для визуализации проблемы.
    Аналогично это можно показывать при наведении.
    В оглавлении по блокам - не выводить, там и так понятно.
1. Привести все остальные заголовки к единообразию, например с Element или Array, могли бы выводить `Array.every` на букву "Е" в алфавите, где его будет искать пользователь.

**Comment by nlopin** [comment]
Спасибо, что подсветил про алфавитный порядок, я совершенно упустил этот момент.

Это еще не финальная структура того, как всё будет выглядеть на релизе. Ближе к запуску я буду приводить содержание в финальный вид. Сейчас мне кажется логичным использовать `.toString()` без указания объекта, к которому метод принадлежит, потому что писать `Number.toString()` неверно — метод принадлежит не самому числу, а прототипу и вызывается на конкретных экземплярах.

Вероятно нужно понять, насколько алфавитный порядок вообще используется, возможно стоит решать эту проблему уже после запуска.
